contributors/design-proposals/pod-pid-namespace.md

```diff
@@ -11,23 +11,67 @@ Runtime Interface and links its rollout in Docker to that of the CRI.
 ## Motivation
 
 Sharing a PID namespace between containers in a pod is discussed in
-[#1615](https://issues.k8s.io/1615), and enables:
+[#1615](https://issues.k8s.io/1615) and enables:
 
-  1. signaling between containers, which is useful for side cars (e.g. for
-     signaling a daemon process after rotating logs).
-  2. easier troubleshooting of pods.
-  3. addressing [Docker's zombie problem][1] by reaping orphaned zombies in the
-     infra container.
+1.  signaling between containers, which is useful for side cars (e.g. for
+    signaling a daemon process after rotating logs).
+1.  easier troubleshooting of pods.
+1.  addressing [Docker's zombie
+    problem](https://blog.phusion.nl/2015/01/20/docker-and-the-pid-1-zombie-reaping-problem/)
+    by reaping orphaned zombies in the infra container.
 
 ## Goals and Non-Goals
 
 Goals include:
-  - Changing default behavior in the Docker runtime as implemented by the CRI
-  - Making Docker behavior compatible with the other Kubernetes runtimes
+
+*   Changing default behavior in the Docker runtime as implemented by the CRI
+*   Making Docker behavior compatible with the other Kubernetes runtimes
 
 Non-goals include:
-  - Creating an init solution that works for all runtimes
-  - Supporting isolated PID namespace indefinitely
+
+*   Creating an general init solution that works for all runtimes
+*   Supporting isolated PID namespace indefinitely
+
+## Requirements
+
+Requirements are listed with the stakeholder who requested it.
+
+*   Must be backwards compatible with every container image that could have run
+    on Kubernetes (i.e. satisfies all implied APIs) [@rhatdan, @smarterclayton]
+
+## Long Term Strategy
+
+In order to avoid breaking compatibility with existing images, the shared
+namespace must be configured per-pod. There are two options for supporting
+per-pod configurable PID namespaces, and the choice hinges mainly on long-term
+plans for PID namespace sharing.
+
+### Option 1: Configurable Namespaces
+
+This option will add first class support for configuring isolated and shared PID
+namespaces for pods. This would be added as CRI requirements and both options
+should be implemented by all runtimes where possible.
+
+### Option 2: Migration to Shared-only Namespaces
+
+In order to avoid breaking compatibility with existing images the default could
+not change until v2.Pod and wouldn't be completed until v1.Pod is deprecated,
+but choosing not to support isolated PID namespaces indefinitely would influence
+design choices. In this scenario we would avoid modifying the CRI or
+implementing isolated namespaces in other runtimes. Enabling the shared
+namespace in v1.Pod could happen in a new field or in an annotation.
```

Comment:
I am having problems parsing the two options. What's the difference? Is it just the default?